controller/sync_namespace.go

```diff
@@ -0,0 +1,47 @@
+package controller
+
+import (
+	"fmt"
+	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/util/argo"
+	gitopscommon "github.com/argoproj/gitops-engine/pkg/sync/common"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func syncNamespace(resourceTracking argo.ResourceTracking, appLabelKey string, trackingMethod v1alpha1.TrackingMethod, appName string, syncPolicy *v1alpha1.SyncPolicy) func(un *unstructured.Unstructured) (bool, error) {
+	return func(un *unstructured.Unstructured) (bool, error) {
```

Comment:
Just to make it more evident what is the `unstructured` resource that we are expecting to receive here I would rename this from `un` to `liveNamespace` or `liveNs` as a shorter alternative.